proposals/Proposal: add cookies.removeAll() method.md

```diff
@@ -0,0 +1,157 @@
+
+## Specification
+
+
+### Schema
+
+##### Syntax
+
+```
+let removed = await browser.cookies.removeAll(
+details // object
+)
+```
+##### Parameters
+An `object` containing information to identify the cookie(s) to remove. It contains the following properties, based on the properties recognized by `cookies.remove`:
+
+>[`name`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#name) optional:
+>
+>A  `string`  representing the name of the cookie to remove.
+>
+>[`partitionKey`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#partitionkey)  optional:
+>
+>An object representing the storage partition containing the cookie.
+>
+>[`topLevelSite`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#toplevelsite)  optional:
+>
+>A  `string`  representing the first-party URL of the top-level site storage partition containing the cookie.
+>
+>[`storeId`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#storeid)  optional:
+>
+>A  `string`  representing the ID of the cookie store to find the cookie in. If unspecified, the cookie is looked for by default in the current execution context's cookie store.
+>
+>[`url`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#url) optional:
+>
+>A  `string`  representing the URL associated with the cookie.
+
+##### Return
+
+> `Promise<Cookie[]>`:
+> 
+> All the cookies that were removed. If list is empty then no cookies were removed.
+
+
+### Behavior
+
+The API will remove all cookies that match the `details` object parameter with the exception of the cases outlined in the implementation details.
+If the extension does not have  [host permissions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions)  for this URL, the API call will fail.
+
+### New Permissions
+No  new  permissions  are  required.
+
+### Manifest File Changes
+No new manifest fields are required.
+
+## Security and Privacy
+Incorrect usage can lead to the removal of cookies that are not intended.
+
+### Exposed Sensitive Data
+None
+
+### Abuse Mitigations
+To remove a cookie, the extension must have host permissions for the `url` associated with the cookie.
+
+
+## Alternatives
+None
+
+### Existing Workarounds
+
+The behavior introduced by this API can be replicated by using a combination of `cookies.getAll()` and `cookies.remove()`.
+
+### Open Web API
+
+The cookies API is specific to extensions
+
+## Implementation Notes
+
+Host permissions are required for this API.
+#### Special cases:
+```cookies.removeAll({})```
+
+>An empty details object will result in an error. 
+
+```
+cookies.removeAll({
+    partitionKey:{}})
+```
+
+>An empty `partitionKey` object as the only value in the details object, will result in all partitioned cookies being removed.
```

Comment:
`partitionKey: {}` in `cookies.getAll` matches all cookies, partitioned and unpartitioned: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/getAll#partitionkey

Can we match that behavior, and if not, why do we deviate from it?

Reply:
My intention was to allow a caller to easily choose between removing just partitioned cookies  or just unpartitioned cookies (by not including a partitionKey). But it keeping the semantics the same between this API and `cookies.getAll` is better for developers as it's one less quirk for them to remember.